### DISCUSSION TITLE
App Router / Auth / Middleware / Cache - What's the best practice to implement protected sites with auth tokens?

### DISCUSSION THREAD

**Discussion by julianhahn** [vercel/next.js#53885] [Help]
### Summary

TDLR: I'm trying to use Supabase as a BAAS and want to implement protected sites with it. In my example we have a public HomePage, a public Login page and a protected /supa page. For the first request to visit the supa page, everything works as expected. The client makes a request to the next server, the middleware hits and checks the route and then checks the cookies from the request, cannot find a valid token and redirects the user to the login. After the login the user is able to visit the protected page. 

Now the part which confuses me: On the protected page I've got a logout button, which deletes the auth cookie and redirects the user to the homepage. But if we click again on the supa link, the NextJS LINK will use the client side cached page for instant navigation and there will be never a request fired for the server middleware to check if the site should be still accessible. Now we got new Cache Documentation but the only sentence I could find which is related to my problem is:
https://nextjs.org/docs/app/building-your-application/caching#router-cache
https://nextjs.org/docs/app/building-your-application/caching#opting-out-3

> It's not possible to opt out of the Router Cache.
> You can opt out of prefetching by setting the prefetch prop of the <Link> component to false. However, this will still temporarily store the route segments for 30s to allow instant navigation between nested segments, such as tab bars, or back and forward navigation. Visited routes will still be cached.

So instead of asking on how to fix my approach, I wanted to ask what is the suggested and best practice approach to implement protected pages with nextjs13? 

(example link is currently broken but will be fixed in a moment)

Thanks already 👍 

### Example

https://jul-mar.de/

**Comment by dlcoffee** [comment]
i have the same issue and used one of the "hacks" proposed here: https://github.com/vercel/next.js/issues/42991#issuecomment-1382024752. the good news is that the team responded recently that they are going to provide a way to configure the router cache.
